File: VB/Demos/08FunctionalProgramming/IPrjService.vb
Imports System.ServiceModel

Namespace FunctionalProgramming
  <ServiceContract>
  Public Interface IPrjService
    <OperationContract(Action:="http://tempuri.org/Divide")>
    Function Divide(intA as integer, intB as Integer) As Integer
    <OperationContract(Action:="http://tempuri.org/Add")>
    Function Add(intA as integer, intB as Integer) As Integer
    <OperationContract(Action:="http://tempuri.org/Multiply")>
    Function Multiply(intA as integer, intB as Integer) As Integer
    <OperationContract(Action:="http://tempuri.org/Subtract")>
    Function Subtract(intA as integer, intB as Integer) As Integer
  End Interface
End Namespace

File: CS/Demos/08FunctionalProgramming/IPrjService.cs
using System.ServiceModel;

namespace Demos.FunctionalProgramming;

[ServiceContract]
public interface IPrjService
{
  [OperationContract(Action = "http://tempuri.org/Divide")]
  int Divide(int intA, int intB);
  [OperationContract(Action = "http://tempuri.org/Add")]
  int Add(int intA, int intB);
  [OperationContract(Action = "http://tempuri.org/Multiply")]
  int Multiply(int intA, int intB);
  [OperationContract(Action = "http://tempuri.org/Subtract")]
  int Subtract(int intA, int intB);
}
